engine/runtime/src/main/java/org/enso/interpreter/service/error/ExitException.java

```diff
@@ -0,0 +1,63 @@
+package org.enso.interpreter.service.error;
+
+import com.oracle.truffle.api.TruffleStackTraceElement;
+import com.oracle.truffle.api.exception.AbstractTruffleException;
+import com.oracle.truffle.api.interop.ExceptionType;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.runtime.data.vector.ArrayLikeHelpers;
+
+/** Thrown when {@code System.exit} call was called during execution. */
+@ExportLibrary(InteropLibrary.class)
+public class ExitException extends AbstractTruffleException implements ServiceException {
+  private final int exitCode;
+
+  public ExitException(int code, Node location) {
+    super("Exit was called with exit code " + code + ".", location);
```

Comment:
The string concatenation deserves to be behind `@TruffleBoundary`... it is because `System` uses the boundary. Wouldn't it be better to have the boundary only when asking for `getExceptionMessage`?

Reply:
Done in 78373a20b0e71ce04a252238d9a8d01fd44c3b5f